examples/camera-app/camera-common/src/camera-app.cpp

```diff
@@ -21,16 +21,60 @@ using namespace chip;
 using namespace chip::app;
 using namespace chip::app::Clusters;
 using namespace chip::app::Clusters::Chime;
+using namespace chip::app::Clusters::WebRTCTransportProvider;
+using namespace chip::app::Clusters::CameraAvStreamManagement;
 
 template <typename T>
 using List   = chip::app::DataModel::List<T>;
 using Status = Protocols::InteractionModel::Status;
 
+CameraApp::CameraApp(chip::EndpointId aClustersEndpoint, CameraDeviceInterface * aCameraDevice)
+{
+    mEndpoint     = aClustersEndpoint;
+    mCameraDevice = aCameraDevice;
+
+    // Instantiate Chime Server
+    mChimeServerPtr = std::make_unique<ChimeServer>(mEndpoint, mCameraDevice->GetChimeDelegate());
+
+    // Instantiate WebRTCTransport Provider
+    mWebRTCTransportProviderPtr =
+        std::make_unique<WebRTCTransportProviderServer>(mCameraDevice->GetWebRTCProviderDelegate(), mEndpoint);
+
+    // Fetch all initialization parameters for CameraAVStreamMgmt Server
+    BitFlags<Feature> features;
+    features.Set(Feature::kSnapshot);
+    BitFlags<OptionalAttribute> optionalAttrs;
+    optionalAttrs.Set(chip::app::Clusters::CameraAvStreamManagement::OptionalAttribute::kNightVision);
+    optionalAttrs.Set(chip::app::Clusters::CameraAvStreamManagement::OptionalAttribute::kNightVisionIllum);
+    uint32_t maxConcurrentVideoEncoders  = mCameraDevice->GetCameraHALInterface().GetMaxConcurrentVideoEncoders();
+    uint32_t maxEncodedPixelRate         = mCameraDevice->GetCameraHALInterface().GetMaxEncodedPixelRate();
+    VideoSensorParamsStruct sensorParams = mCameraDevice->GetCameraHALInterface().GetVideoSensorParams();
```

Comment:
Those initialization are not platform independent, should be moved to linux

Reply:
These are accessing the abstract Camera HAL Interface functions. The implementation of those functions would be platform-specific.

Reply:
Acked